AssertionError: 2 != 0 : Not all expected timeslots deleted

### Describe the issue

A lot of CI runs lately are failing like this:
```
======================================================================
FAIL: test_delete_time_interval (ietf.meeting.tests_js.EditTimeslotsTests)
Delete button for a time interval should delete all timeslots in that interval
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/datatracker/datatracker/ietf/meeting/tests_js.py", line 1667, in test_delete_time_interval
    self.do_delete_time_interval_test(cancel=False)
  File "/__w/datatracker/datatracker/ietf/meeting/tests_js.py", line 1663, in do_delete_time_interval_test
    self.do_delete_test(selector, keep, delete, cancel)
  File "/__w/datatracker/datatracker/ietf/meeting/tests_js.py", line 1594, in do_delete_test
    self.assertEqual(
AssertionError: 2 != 0 : Not all expected timeslots deleted
```
See https://github.com/ietf-tools/datatracker/actions/runs/3581847477/jobs/6025382408

Do we know why and can this be fixed?

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

--- a/ietf/meeting/tests_js.py
+++ b/ietf/meeting/tests_js.py
@@ -1657,9 +1657,6 @@ class EditTimeslotsTests(IetfSeleniumTestCase):
 
     def test_delete_time_interval(self):
         """Delete button for a time interval should delete all timeslots in that interval"""
-        self.meeting.time_zone = 'Etc/GMT-11'
-        self.meeting.save()
-        debug.show('self.meeting.time_zone')
         self.do_delete_time_interval_test(cancel=False)
 
     def test_delete_time_interval_cancel(self):

--- a/ietf/meeting/tests_js.py
+++ b/ietf/meeting/tests_js.py
@@ -1621,30 +1621,27 @@ class EditTimeslotsTests(IetfSeleniumTestCase):
         self.do_delete_timeslot_test(cancel=True)
 
     def do_delete_time_interval_test(self, cancel=False):
-        delete_day = self.meeting.date
-        delete_time = datetime.time(hour=10)
-        other_day = self.meeting.get_meeting_date(1)
-        other_time = datetime.time(hour=12)
+        delete_time_local = datetime_from_date(self.meeting.date, self.meeting.tz()).replace(hour=10)
+        delete_time = delete_time_local.astimezone(datetime.timezone.utc)
         duration = datetime.timedelta(minutes=60)
 
         delete: [TimeSlot] = TimeSlotFactory.create_batch(
             2,
             meeting=self.meeting,
-            time=datetime_from_date(delete_day, self.meeting.tz()).replace(hour=delete_time.hour),
+            time=delete_time_local,
             duration=duration,
         )
-
         keep: [TimeSlot] = [
             TimeSlotFactory(
                 meeting=self.meeting,
-                time=datetime_from_date(day, self.meeting.tz()).replace(hour=time.hour),
+                time=keep_time,
                 duration=duration
             )
-            for (day, time) in (
-                # combinations of day/time that should not be deleted
-                (delete_day, other_time),
-                (other_day, delete_time),
-                (other_day, other_time),
+            for keep_time in (
+                # same day, but 2 hours later
+                delete_time + datetime.timedelta(hours=2),
+                # next day, but same wall clock time
+                datetime_from_date(self.meeting.get_meeting_date(1), self.meeting.tz()).replace(hour=10),
             )
         ]
 
@@ -1652,18 +1649,17 @@ class EditTimeslotsTests(IetfSeleniumTestCase):
             '#timeslot-table '
             '.delete-button[data-delete-scope="column"]'
             '[data-col-id="{}T{}-{}"]'.format(
-                delete_day.isoformat(),
-                delete_time.strftime('%H:%M'),
-                self.meeting.tz().localize(
-                    datetime.datetime.combine(delete_day, delete_time) + duration
-                ).strftime(
-                    '%H:%M'
-                ))
+                delete_time_local.date().isoformat(),
+                delete_time_local.strftime('%H:%M'),
+                (delete_time + duration).astimezone(self.meeting.tz()).strftime('%H:%M'))
         )
         self.do_delete_test(selector, keep, delete, cancel)
 
     def test_delete_time_interval(self):
         """Delete button for a time interval should delete all timeslots in that interval"""
+        self.meeting.time_zone = 'Etc/GMT-11'
+        self.meeting.save()
+        debug.show('self.meeting.time_zone')
         self.do_delete_time_interval_test(cancel=False)
 
     def test_delete_time_interval_cancel(self):

--- a/ietf/templates/meeting/timeslot_edit_timeslot.html
+++ b/ietf/templates/meeting/timeslot_edit_timeslot.html
@@ -1,8 +1,8 @@
 <div id="timeslot{{ ts.pk }}"
     class="timeslot {% if ts in in_official_use %}in-official-use{% elif ts in in_use %}in-unofficial-use{% endif %}"
     data-timeslot-pk="{{ ts.pk }}"
-    data-date-id="{{ ts.time.date.isoformat }}"{# used for identifying day/col contents #}
-    data-col-id="{{ ts.time.date.isoformat }}T{{ ts.time|date:"H:i" }}-{{ ts.end_time|date:"H:i" }}" {# used for identifying column contents #}
+    data-date-id="{{ ts.local_start_time.date.isoformat }}"{# used for identifying day/col contents #}
+    data-col-id="{{ ts.local_start_time.date.isoformat }}T{{ ts.time|date:"H:i" }}-{{ ts.end_time|date:"H:i" }}" {# used for identifying column contents #}
 data-timeslot-name="{{ ts.name }}"
 data-timeslot-type="{{ ts.type.slug }}"
 data-timeslot-location="{{ ts.location.name }}"

